signac/core/synced_collection.py

```diff
@@ -50,8 +54,13 @@ def register(cls, *args):
         """
         if not hasattr(cls, 'registry'):
             cls.registry = defaultdict(list)
+        if not hasattr(cls, 'backend_registry'):
+            cls.backend_registry = []
         for _cls in args:
-            cls.registry[_cls.backend].append(_cls)
+            if not _cls.__abstractmethods__:
```

Comment:
Use the standard library's `inspect` module to determine if the class is abstract (i.e. has abstract methods). Put `import inspect` at the top.
```suggestion
            if not inspect.isabstract(_cls):
```